internal/alertmanager/currentalert/process.go

```diff
@@ -262,20 +263,26 @@ func resizePvc(log logr.Logger, labels model.LabelSet, annotiations model.LabelS
 
 			storageConfigs := brokerConfig.StorageConfigs
 
-			for n, c := range storageConfigs {
+			for _, c := range storageConfigs {
 				modifiableConfig := c.DeepCopy()
 				if modifiableConfig.MountPath == pvc.Annotations["mountPath"] {
 					size := *modifiableConfig.PvcSpec.Resources.Requests.Storage()
 					size.Add(incrementBy)
 
 					modifiableConfig.PvcSpec.Resources.Requests["storage"] = size
 
+					// When the storage is in brokerConfigGroup we don't resize the storage there because in that case
+					// all of those brokers that are using this brokerConfigGroup its storage would be resized.
+					// We add the storage into the broker.BrokerConfig.StorageConfigs in this way only that broker pvc will be resized.
 					if broker.BrokerConfig == nil {
-						broker.BrokerConfig = &v1beta1.BrokerConfig{
-							StorageConfigs: []v1beta1.StorageConfig{*modifiableConfig},
-						}
+						broker.BrokerConfig = &v1beta1.BrokerConfig{}
```

Comment:
**Question**: in this case shouldn't this instead be creating a broker specific storage config with the new size on a new PV? 
Like we do with brokerConfig, but mountPath index == -1 case.
Or is that unfeasible in this case?

Reply:
We discussed it offline that in this case we do not want to do that, because we cannot determine what exactly to copy over from the brokerConfigGroup to the broker specific config and what to leave, the merging of the two is not simple enough so there is no todo with this.